pkg/workflow/couler/template.go

```diff
@@ -56,7 +57,7 @@ if "{{.SecretName}}" != "":
 
 couler.clean_workflow_after_seconds_finished({{.WorkflowTTL}})
 def escape_sql(original_sql):
-	return original_sql.replace('"', '\\"')
+    return re.sub(r'(["$` + "`" + `\\])', r'\\\1', original_sql)
```

Comment:
Add a sqlflow Couler step can avoid backtick in backtick. https://github.com/sql-machine-learning/sqlflow/pull/2060